pandas/_testing.py

```diff
@@ -1176,20 +1177,23 @@ def assert_series_equal(
             raise AssertionError(msg)
     elif is_interval_dtype(left.dtype) or is_interval_dtype(right.dtype):
         assert_interval_array_equal(left.array, right.array)
-    elif is_datetime64tz_dtype(left.dtype):
-        # .values is an ndarray, but ._values is the ExtensionArray.
+    elif is_categorical_dtype(left.dtype) or is_categorical_dtype(right.dtype):
+        _testing.assert_almost_equal(
+            left._values,
+            right._values,
+            check_less_precise=check_less_precise,
+            check_dtype=check_dtype,
+            obj=str(obj),
+        )
+    elif is_extension_array_dtype(left.dtype) or is_extension_array_dtype(right.dtype):
```

Comment:
Before this was "and", and now it is "or". That seems to cause issues in geopandas in case you specify `check_dtype=False` (in which case you can eg have an EA dtype and object dtype with equal values)